// ram-product-list/esm5/lib/product-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
var ProductListComponent = /** @class */ (function () {
    function ProductListComponent() {
        this.name = 'test';
        this.items = [];
    }
    /**
     * @return {?}
     */
    ProductListComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    ProductListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ram-product-list',
                    template: "<p>{{name}} List</p>\n<table class=\"table table-hover\">\n    <thead>\n        <tr>\n            <th class=\"col-size-sm\">#</th>\n            <th class=\"col-size-lg\">Code</th>\n            <th class=\"col-size-lg\">Name</th>\n            <th>About</th>\n            <th>Price ($)</th>\n            <th>Raiting</th>\n        </tr>\n    </thead>\n    <tbody>\n        <ng-container *ngIf=\"items?.length else noData\">\n            <tr *ngFor=\"let item of items; let i = index\">\n                <td class=\"col-size-sm\">{{i+1}}/{{items.length}}</td>\n                <td class=\"col-size-lg\">{{item.productCode}}</td>\n                <td class=\"col-size-lg\">{{item.productName}}</td>\n                <td>{{item.description}}</td>\n                <td style=\"text-align: center\">{{item.value | number : '1.0-2' }}</td>\n                <td style=\"text-align: center\">{{item.starRating}}</td>\n            </tr>\n        </ng-container>\n    </tbody>\n</table>\n\n<ng-template #noData>\n    <div class=\"mx-3 my-3\">\n        No Data available.\n    </div>\n</ng-template>",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    ProductListComponent.ctorParameters = function () { return []; };
    ProductListComponent.propDecorators = {
        "name": [{ type: Input },],
        "items": [{ type: Input },],
    };
    return ProductListComponent;
}());
export { ProductListComponent };
function ProductListComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ProductListComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ProductListComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    ProductListComponent.propDecorators;
    /** @type {?} */
    ProductListComponent.prototype.name;
    /** @type {?} */
    ProductListComponent.prototype.items;
}
