// ram-product-list/bundles/ram-product-list.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/platform-browser')) :
    typeof define === 'function' && define.amd ? define('ram-product-list', ['exports', '@angular/core', '@angular/platform-browser'], factory) :
    (factory((global['ram-product-list'] = {}),global.ng.core,global.ng.platformBrowser));
}(this, (function (exports,i0,platformBrowser) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var ProductListService = (function () {
        function ProductListService() {
        }
        ProductListService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        /** @nocollapse */
        ProductListService.ctorParameters = function () { return []; };
        /** @nocollapse */ ProductListService.ngInjectableDef = i0.defineInjectable({ factory: function ProductListService_Factory() { return new ProductListService(); }, token: ProductListService, providedIn: "root" });
        return ProductListService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var ProductListComponent = (function () {
        function ProductListComponent() {
            this.name = 'test';
            this.items = [];
        }
        /**
         * @return {?}
         */
        ProductListComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        ProductListComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ram-product-list',
                        template: "<p>{{name}} List</p>\n<table class=\"table table-hover\">\n    <thead>\n        <tr>\n            <th class=\"col-size-sm\">#</th>\n            <th class=\"col-size-lg\">Code</th>\n            <th class=\"col-size-lg\">Name</th>\n            <th>About</th>\n            <th>Price ($)</th>\n            <th>Raiting</th>\n        </tr>\n    </thead>\n    <tbody>\n        <ng-container *ngIf=\"items?.length else noData\">\n            <tr *ngFor=\"let item of items; let i = index\">\n                <td class=\"col-size-sm\">{{i+1}}/{{items.length}}</td>\n                <td class=\"col-size-lg\">{{item.productCode}}</td>\n                <td class=\"col-size-lg\">{{item.productName}}</td>\n                <td>{{item.description}}</td>\n                <td style=\"text-align: center\">{{item.value | number : '1.0-2' }}</td>\n                <td style=\"text-align: center\">{{item.starRating}}</td>\n            </tr>\n        </ng-container>\n    </tbody>\n</table>\n\n<ng-template #noData>\n    <div class=\"mx-3 my-3\">\n        No Data available.\n    </div>\n</ng-template>",
                        styles: []
                    },] },
        ];
        /** @nocollapse */
        ProductListComponent.ctorParameters = function () { return []; };
        ProductListComponent.propDecorators = {
            "name": [{ type: i0.Input },],
            "items": [{ type: i0.Input },],
        };
        return ProductListComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var ProductListModule = (function () {
        function ProductListModule() {
        }
        ProductListModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [platformBrowser.BrowserModule],
                        declarations: [ProductListComponent],
                        exports: [ProductListComponent]
                    },] },
        ];
        return ProductListModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.ProductListService = ProductListService;
    exports.ProductListComponent = ProductListComponent;
    exports.ProductListModule = ProductListModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
