// ram-product-list/fesm5/ram-product-list.js
import { Injectable, Component, Input, NgModule, defineInjectable } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var ProductListService = /** @class */ (function () {
    function ProductListService() {
    }
    ProductListService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    ProductListService.ctorParameters = function () { return []; };
    /** @nocollapse */ ProductListService.ngInjectableDef = defineInjectable({ factory: function ProductListService_Factory() { return new ProductListService(); }, token: ProductListService, providedIn: "root" });
    return ProductListService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var ProductListComponent = /** @class */ (function () {
    function ProductListComponent() {
        this.name = 'test';
        this.items = [];
    }
    /**
     * @return {?}
     */
    ProductListComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    ProductListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ram-product-list',
                    template: "<p>{{name}} List</p>\n<table class=\"table table-hover\">\n    <thead>\n        <tr>\n            <th class=\"col-size-sm\">#</th>\n            <th class=\"col-size-lg\">Code</th>\n            <th class=\"col-size-lg\">Name</th>\n            <th>About</th>\n            <th>Price ($)</th>\n            <th>Raiting</th>\n        </tr>\n    </thead>\n    <tbody>\n        <ng-container *ngIf=\"items?.length else noData\">\n            <tr *ngFor=\"let item of items; let i = index\">\n                <td class=\"col-size-sm\">{{i+1}}/{{items.length}}</td>\n                <td class=\"col-size-lg\">{{item.productCode}}</td>\n                <td class=\"col-size-lg\">{{item.productName}}</td>\n                <td>{{item.description}}</td>\n                <td style=\"text-align: center\">{{item.value | number : '1.0-2' }}</td>\n                <td style=\"text-align: center\">{{item.starRating}}</td>\n            </tr>\n        </ng-container>\n    </tbody>\n</table>\n\n<ng-template #noData>\n    <div class=\"mx-3 my-3\">\n        No Data available.\n    </div>\n</ng-template>",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    ProductListComponent.ctorParameters = function () { return []; };
    ProductListComponent.propDecorators = {
        "name": [{ type: Input },],
        "items": [{ type: Input },],
    };
    return ProductListComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var ProductListModule = /** @class */ (function () {
    function ProductListModule() {
    }
    ProductListModule.decorators = [
        { type: NgModule, args: [{
                    imports: [BrowserModule],
                    declarations: [ProductListComponent],
                    exports: [ProductListComponent]
                },] },
    ];
    return ProductListModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { ProductListService, ProductListComponent, ProductListModule };
