// ram-product-list/esm2015/lib/product-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class ProductListComponent {
    constructor() {
        this.name = 'test';
        this.items = [];
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
ProductListComponent.decorators = [
    { type: Component, args: [{
                selector: 'ram-product-list',
                template: `<p>{{name}} List</p>
<table class="table table-hover">
    <thead>
        <tr>
            <th class="col-size-sm">#</th>
            <th class="col-size-lg">Code</th>
            <th class="col-size-lg">Name</th>
            <th>About</th>
            <th>Price ($)</th>
            <th>Raiting</th>
        </tr>
    </thead>
    <tbody>
        <ng-container *ngIf="items?.length else noData">
            <tr *ngFor="let item of items; let i = index">
                <td class="col-size-sm">{{i+1}}/{{items.length}}</td>
                <td class="col-size-lg">{{item.productCode}}</td>
                <td class="col-size-lg">{{item.productName}}</td>
                <td>{{item.description}}</td>
                <td style="text-align: center">{{item.value | number : '1.0-2' }}</td>
                <td style="text-align: center">{{item.starRating}}</td>
            </tr>
        </ng-container>
    </tbody>
</table>

<ng-template #noData>
    <div class="mx-3 my-3">
        No Data available.
    </div>
</ng-template>`,
                styles: []
            },] },
];
/** @nocollapse */
ProductListComponent.ctorParameters = () => [];
ProductListComponent.propDecorators = {
    "name": [{ type: Input },],
    "items": [{ type: Input },],
};
function ProductListComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ProductListComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ProductListComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    ProductListComponent.propDecorators;
    /** @type {?} */
    ProductListComponent.prototype.name;
    /** @type {?} */
    ProductListComponent.prototype.items;
}
