// ram-product-list/fesm2015/ram-product-list.js
import { Injectable, Component, Input, NgModule, defineInjectable } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ProductListService {
    constructor() { }
}
ProductListService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
ProductListService.ctorParameters = () => [];
/** @nocollapse */ ProductListService.ngInjectableDef = defineInjectable({ factory: function ProductListService_Factory() { return new ProductListService(); }, token: ProductListService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ProductListComponent {
    constructor() {
        this.name = 'test';
        this.items = [];
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
ProductListComponent.decorators = [
    { type: Component, args: [{
                selector: 'ram-product-list',
                template: `<p>{{name}} List</p>
<table class="table table-hover">
    <thead>
        <tr>
            <th class="col-size-sm">#</th>
            <th class="col-size-lg">Code</th>
            <th class="col-size-lg">Name</th>
            <th>About</th>
            <th>Price ($)</th>
            <th>Raiting</th>
        </tr>
    </thead>
    <tbody>
        <ng-container *ngIf="items.length; else loading">
            <tr *ngFor="let item of items; let i = index">
                <td class="col-size-sm">{{i+1}}/{{items.length}}</td>
                <td class="col-size-lg">{{item.productCode}}</td>
                <td class="col-size-lg">{{item.productName}}</td>
                <td>{{item.description}}</td>
                <td style="text-align: center">{{item.value | number : '1.0-2' }}</td>
                <td style="text-align: center">{{item.starRating}}</td>
            </tr>
        </ng-container>
    </tbody>
</table>
<ng-template #loading>No Data.</ng-template>`,
                styles: []
            },] },
];
/** @nocollapse */
ProductListComponent.ctorParameters = () => [];
ProductListComponent.propDecorators = {
    "name": [{ type: Input },],
    "items": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ProductListModule {
}
ProductListModule.decorators = [
    { type: NgModule, args: [{
                imports: [BrowserModule],
                declarations: [ProductListComponent],
                exports: [ProductListComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { ProductListService, ProductListComponent, ProductListModule };
